include/seqan3/range/container/constexpr_vector.hpp

```diff
@@ -0,0 +1,919 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \brief A constexpr vector implementation with dynamic size at compile time.
+ */
+
+#pragma once
+
+#include <array>
+#include <type_traits>
+
+#include <range/v3/view/repeat_n.hpp>
+
+#if SEQAN3_WITH_CEREAL
```

Comment:
Did we decide on how we want to handle the cereal includes?

Reply:
they will stay in the header where they are needed (like it is now).
We will discuss at the next strategy meeting if we should make cereal a hard dependency. Then we can remove all this HAS_CEREAL stuff